torch/distributed/_tools/fsdp2_mem_tracker.py

```diff
@@ -185,8 +200,8 @@ def _instrument_fsdp_sharded_params_grads(
     def _fsdp_state_pre_forward(
         self,
         fsdp_mod: FSDPModule,
-        orig_fsdp_state_pre_fw: Callable,
-    ) -> Callable:
+        orig_fsdp_state_pre_fw: Callable[_P, Any],
+    ) -> Callable[_P, Tuple[Tuple[Any, ...], Dict[str, Any]]]:
```

Comment:
Why tuple and dict any here?

Reply:
That is just copied from the return type of the nested wrapper function which had that type.

Reply:
Shouldn't this actually be the return type of `orig_fsdp_state_pre_fw`? That's what is actually returned (an the inner function should be accordingly updated...

Reply:
It kind of is already – just with the quirk that the return value of `orig_fsdp_state_pre_fw` is unpacked in the inner function and then the tuple is returned. So in this case a plain use of e.g. `TypeVar` for the return value would raise an error.

But now with this comment I noticed that in this signature we could replace the current return type `Any` of `orig_fsdp_state_pre_fw` with this tuple type as well. We could also use `TypeVarTuple` as before to replace the `Any, ...` to preserve more information. How does that sound? The types would look as follows

    def _fsdp_state_pre_forward(
        self,
        fsdp_mod: FSDPModule,
        orig_fsdp_state_pre_fw: Callable[_P, Tuple[Tuple[Unpack[_Ts]], Dict[str, Any]]],
    ) -> Callable[_P, Tuple[Tuple[Unpack[_Ts]], Dict[str, Any]]]:
        @wraps(orig_fsdp_state_pre_fw)
        def inner(
            *args: _P.args, **kwargs: _P.kwargs
        ) -> Tuple[Tuple[Unpack[_Ts]], Dict[str, Any]]:
            ...
            args, kwargs = orig_fsdp_state_pre_fw(*args, **kwargs)
            ...
            return args, kwargs

        return inner

Reply:
Assuming that works @kaspell, looks good to me.